### DISCUSSION TITLE
Disk and memory exceptions

### DISCUSSION THREAD

**Discussion by XTUxiongda** [minio/minio#16776] [Q&A] [answered]
Our deployment method is 4 physical machines with 128G memory, and each host hangs 4 pieces of 2T disks
Recently, it was discovered that the memory usage has been above 50%, and many swap partitions have been used
Our disks are using SAS, not SSD
Will Minio use memory as a cache layer?
Recently, there is not much data volume on the minio cluster, but sporadic upload and download time-consuming surges often occur, and the duration is not long.
The monitoring data shows that the disk utilization rate is relatively high, and I feel that the disk IO is the bottleneck, but the cluster is newly built, and there is not much traffic coming in. This has puzzled me for a long time, and I hope to get help

**Comment by klauspost** [comment, marked as answer]
MinIO doesn't have in-memory cache.

We offer setup/performance reviews as part of our [subscription offers](https://min.io/pricing).
